gnd/src/main/java/com/google/android/gnd/ui/editobservation/MultiSelectDialogFactory.java

```diff
@@ -16,84 +16,75 @@
 
 package com.google.android.gnd.ui.editobservation;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-import static java8.util.stream.StreamSupport.stream;
+import static com.google.android.gnd.util.ImmutableListCollector.toImmutableList;
+import static java.util.Objects.requireNonNull;
 
 import android.content.Context;
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.AlertDialog;
-import com.google.android.gnd.R;
-import com.google.android.gnd.model.form.Field;
 import com.google.android.gnd.model.form.MultipleChoice;
 import com.google.android.gnd.model.form.Option;
 import com.google.android.gnd.model.observation.MultipleChoiceResponse;
 import com.google.android.gnd.model.observation.Response;
+import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
-import java.util.List;
 import java8.util.Optional;
 import java8.util.function.Consumer;
 import java8.util.stream.IntStreams;
 
-// TODO: Replace with modal bottom sheet.
-class MultiSelectDialogFactory {
+@AutoValue
+abstract class MultiSelectDialogFactory extends SelectDialogFactory {
 
-  private Context context;
+  @Nullable private boolean[] checkedItems;
```

Comment:
The size is determined after calling `initSelectedState()`. Also, `setMultiChoiceItems` also expects `null` if none of the items are selected. We can also use a collection instead of an array. But then we'll have to manually track all selected items. This way, the original array gets updated which we directly use.